lxc/image.go

```diff
@@ -389,41 +388,30 @@ func (c *imageCmd) run(conf *config.Config, args []string) error {
 			return err
 		}
 
-		// Optimisation for simplestreams
-		var imgInfo *api.Image
+		// Attempt to resolve an image alias
 		image := inName
-		if conf.Remotes[remote].Protocol == "simplestreams" {
-			imgInfo = &api.Image{}
-			imgInfo.Fingerprint = image
-			imgInfo.Public = true
-		} else {
-
-			// Attempt to resolve an image alias
-			alias, _, err := d.GetImageAlias(image)
-			if err == nil {
-				image = alias.Target
-			}
+		alias, _, err := d.GetImageAlias(image)
```

Comment:
That's just as expensive as GetImage() unfortunately. Anything which actually requires reading simplestreams data will cause the 3M large json index to be downloaded and parse.

GetImageAlias shouldn't be needed when dealing with simplestreams without the copy-alias bit.